temporaltest/server_test.go

```diff
@@ -0,0 +1,407 @@
+// The MIT License
+//
+// Copyright (c) 2021 Datadog, Inc.
+//
+// Copyright (c) 2020 Temporal Technologies Inc.  All rights reserved.
+//
+// Copyright (c) 2020 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package temporaltest_test
+
+import (
+	"strings"
+	"testing"
+	"time"
+
+	"context"
+	"fmt"
+
+	"go.temporal.io/api/enums/v1"
+	"go.temporal.io/api/operatorservice/v1"
+	"go.temporal.io/sdk/activity"
+	"go.temporal.io/sdk/client"
+	"go.temporal.io/sdk/interceptor"
+	"go.temporal.io/sdk/worker"
+	"go.temporal.io/sdk/workflow"
+
+	"go.temporal.io/server/common/authorization"
+	"go.temporal.io/server/common/config"
+	"go.temporal.io/server/temporal"
+	"go.temporal.io/server/temporaltest"
+)
+
+// to be used in example code
+var t *testing.T
+
+func ExampleNewServer() {
+	// Create test Temporal server and client
+	ts := temporaltest.NewServer(temporaltest.WithT(t))
+	c := ts.GetDefaultClient()
+
+	// Register a new worker on the `hello_world` task queue
+	ts.NewWorker("hello_world", func(registry worker.Registry) {
+		RegisterWorkflowsAndActivities(registry)
+	})
+
+	// Start test workflow
+	wfr, err := c.ExecuteWorkflow(
+		context.Background(),
+		client.StartWorkflowOptions{TaskQueue: "hello_world"},
+		Greet,
+		"world",
+	)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Get workflow result
+	var result string
+	if err := wfr.Get(context.Background(), &result); err != nil {
+		t.Fatal(err)
+	}
+
+	// Print result
+	fmt.Println(result)
+	// Output: Hello world
+}
+
+func TestNewServer(t *testing.T) {
```

Comment:
can you hook these up in the Makefile so they run automatically? probably with the "integration" tests?

Reply:
I believe they're already running as part of `make unit-test`. You'd prefer them to be included in the integration tests instead?

Reply:
(also commented above) yeah, I think integration makes more sense since they do run a whole server

Reply:
I think these tests should now only run with integration. I've added `internal/temporalite` to integration as well (even though there aren't any tests in that package at the moment).